programs/zstdcli.c

```diff
@@ -1197,15 +1199,22 @@ int main(int const argCount, const char* argv[])
     /* IO Stream/File */
     FIO_setNotificationLevel(g_displayLevel);
     FIO_setPatchFromMode(prefs, patchFromDictFileName != NULL);
-    if (patchFromDictFileName != NULL) {
-        dictFileName = patchFromDictFileName;
-    }
     if (memLimit == 0) {
         if (compressionParams.windowLog == 0) {
             memLimit = (U32)1 << g_defaultMaxWindowLog;
         } else {
             memLimit = (U32)1 << (compressionParams.windowLog & 31);
     }   }
+    if (patchFromDictFileName != NULL) {
+        const unsigned long long dictSize = UTIL_getFileSize(patchFromDictFileName);
+        if (dictSize != UTIL_FILESIZE_UNKNOWN) {
+            memLimit = dictSize > memLimit ? (unsigned)dictSize : memLimit;
```

Comment:
Doesn't `memLimit` have to be `fileSize`, not `dictSize`? The dictionary will stay in-bounds as long as a single byte of the dictionary is in-bounds, so the `memLimit` should be the file size.